poc/src/main/scala/poc/frameworks/dao/MockUserDao.scala

```diff
@@ -0,0 +1,37 @@
+package poc.frameworks.dao
+
+import poc.frameworks.model.internal.User
+
+import scala.collection.mutable.ListBuffer
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+/**
+  * Created by arunavas on 28/9/17.
+  */
```

Comment:
@apoorvingle, please allow the mock part for now as this is just a poc, focused mainly on the application flow. I have noted your suggestion on using property based testing instead mocking and will surely use that in the actual code :)